File: AnagramChecker/anagram_checker.rb
def to_counts(str)
  letter_counts = Hash.new(0)

  str.chars.each { |char| letter_counts[char] += 1 }

  letter_counts
end

def is_anagram?(str1, str2)
  to_counts(str1) == to_counts(str2)
end

if __FILE__ == $PROGRAM_NAME
  puts 'Enter a word:'
  print '>> '
  s1 = gets.strip
  puts 'Enter another word:'
  print '>> '
  s2 = gets.strip

  puts is_anagram?(s1, s2) ? "#{s1} and #{s2} are anagrams!" : "#{s1} and #{s2} are not anagrams!"
end

File: AnagramChecker/AnagramChecker.java
import java.util.Arrays;
import java.util.Scanner;

public class AnagramChecker
{
	public static void main(String[] args)
	{
		Scanner scanner = new Scanner(System.in);
		System.out.print("Enter a word: ");
		String s1 = scanner.nextLine();
		System.out.print("Enter another word: ");
		String s2 = scanner.nextLine();

		if(isAnagram(s1, s2))
			System.out.println(s1 + " and " + s2 + " are anagrams");
		else
			System.out.println(s1 + " and " + s2 + " are not anagrams");

		scanner.close();
	}

	public static boolean isAnagram(String s1, String s2)
	{
		if(s1.length()!=s2.length())return false;
		char [] c1 = s1.toCharArray();
		char [] c2 = s2.toCharArray();
		Arrays.sort(c1);
		arrays.sort(c2);
		String a1 = "", a2 = "";
		for(int i=0;i<c1.length; i++){
			a1+=c1[i];
			a2+=c2[i];
		}
		if(a1.equals(a2))return true;
		return false;
	}
}
